meta-facebook/yv4-sd/src/platform/plat_mctp.c

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "plat_mctp.h"
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <logging/log.h>
+#include <logging/log_ctrl.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include "mctp.h"
+#include "mctp_ctrl.h"
+#include "pldm.h"
+#include "ipmi.h"
+#include "sensor.h"
+#include "plat_ipmb.h"
+
+#include "hal_i2c.h"
+
+LOG_MODULE_REGISTER(plat_mctp);
+
+#define MCTP_MSG_TYPE_SHIFT 0
+#define MCTP_MSG_TYPE_MASK 0x7F
+#define MCTP_IC_SHIFT 7
+#define MCTP_IC_MASK 0x80
+
+/* i2c 8 bit address */
+#define I2C_ADDR_BIC 0x40
+#define I2C_ADDR_BMC 0x20
+
+/* i2c dev bus*/
+#define I2C_BUS_BMC 0x02
+
+/* mctp endpoint */
+#define MCTP_EID_BMC 0x09
+#define MCTP_EID_SELF 0x0A
+
+K_TIMER_DEFINE(send_cmd_timer, send_cmd_to_dev, NULL);
+K_WORK_DEFINE(send_cmd_work, send_cmd_to_dev_handler);
+
+typedef struct _mctp_smbus_port {
+	mctp *mctp_inst;
+	mctp_medium_conf conf;
+	uint8_t user_idx;
+} mctp_smbus_port;
+
+/* mctp route entry struct */
+typedef struct _mctp_route_entry {
+	uint8_t endpoint;
+	uint8_t bus; /* TODO: only consider smbus/i3c */
+	uint8_t addr; /* TODO: only consider smbus/i3c */
+} mctp_route_entry;
+
+typedef struct _mctp_msg_handler {
+	MCTP_MSG_TYPE type;
+	mctp_fn_cb msg_handler_cb;
+} mctp_msg_handler;
+
+static mctp_smbus_port smbus_port[] = {
+	{ .conf.smbus_conf.addr = I2C_ADDR_BIC, .conf.smbus_conf.bus = I2C_BUS_BMC },
+};
+
+mctp_route_entry mctp_route_tbl[] = {
+	{ MCTP_EID_BMC, I2C_BUS_BMC, I2C_ADDR_BMC },
+};
+
+static mctp *find_mctp_by_smbus(uint8_t bus)
+{
+	uint8_t i;
+	for (i = 0; i < ARRAY_SIZE(smbus_port); i++) {
+		mctp_smbus_port *p = smbus_port + i;
+
+		if (bus == p->conf.smbus_conf.bus) {
+			return p->mctp_inst;
+		}
+	}
+
+	return NULL;
+}
+
+static void set_endpoint_resp_handler(void *args, uint8_t *buf, uint16_t len)
+{
+	ARG_UNUSED(args);
+	CHECK_NULL_ARG(buf);
+
+	LOG_HEXDUMP_DBG(buf, len, __func__);
+}
+
+static void set_endpoint_resp_timeout(void *args)
+{
+	CHECK_NULL_ARG(args);
```

Comment:
The API is not setting any timeout or are you going to populate it in next commit? If so, better to leave a TODO for indications.

Reply:
It doesn't set device endpoint in this commit. I leaved a TODO for indication.